va/va_dec_jpeg.h

```diff
@@ -75,8 +75,10 @@ typedef struct _VAPictureParameterBufferJPEGBaseline {
     uint8_t       color_space;
     /** \brief Set to VA_ROTATION_* for a single rotation angle reported by VAConfigAttribDecJPEG. */
     uint32_t      rotation;
+    /** \brief region-of-interest boundary in pixels */
+    VARectangle   roi_rectangle;
```

Comment:
it is the ROI rectangle boundary in pixels , zero is to be ignored and considered as no ROI decode request and process complete image decode. And it is single rectangle ROI support, no support for multiple ROI rectangles.

Reply:
thanks, but still  have some questions:
1. sound like it is something like "crop" ?, or jpeg spec support decode from any block?
2.  how about the output surfaces?  if I create a full size surface, it means only part data in rectangle is filled at the left top corner?

btw: which platform support this feature?

Reply:
1. Yes you are right it is like crop or decode of any specific block defined by roi_rectangle.
2. Application is supposed to allocate output surface with resolution that exactly match the roi_rectangle it requested for decode. Output surface properties (resolution) will be validated across the roi_rectangle definition to make sure the output surface is the right size for the requested roi_rectangle.

Upcoming AMD platform supports this feature.

Reply:
> 2. Application is supposed to allocate output surface with resolution that exactly match the roi_rectangle it requested for decode. Output surface properties (resolution) will be validated across the roi_rectangle definition to make sure the output surface is the right size for the requested roi_rectangle.

it is hard request or recommended one?  my question is: any resolution bigger than roi_rectangle is ok to be output?

Reply:
it is a hard request that surface size has to match with actual roi_rectangle request size , bigger resolution of surface will create surface pitch related issues.

Reply:
Also hard request because, exact surface size (matching roi_rectangle) is the render target which can be submitted to renderer/display or submit to post process on gpu, bigger size surface will not be right choice for this pupose as well.

Reply:
from the libva perspective, there are 3 ways to achieve similar goal.
1. decode call + vpp(crop) call
2. decode call + VAConfigAttribDecProcessing in one call
3. this interface for jpeg only, rectangle should be aligned with block size (8x8), but the call itself is simple. 

this PR LGTM, maybe using different name "crop_xxx" is better?

Reply:
okay, thank you for explaining and summarizing.

Sure, will change the bitfield variable as "crop" and the VARectangle  variable as "crop_rectangle" ?
and will appropriately change the commit message also to

jpeg: add support for crop and partial decode